src/libDirectoryService/DirectoryService.cpp

```diff
@@ -488,6 +489,38 @@ bool DirectoryService::FinishRejoinAsDS() {
   }
 
   LOG_MARKER();
+  for (auto& i : *m_mediator.m_DSCommittee) {
+    if (i.first == m_mediator.m_selfKey.second) {
+      i.second = Peer();
+      LOG_GENERAL(
+          INFO,
+          "Found current node to be inside ds committee. Setting it to Peer()");
+      break;
+    }
+  }
+
+  LOG_GENERAL(INFO, "DS committee is ");
+  for (const auto& i : *m_mediator.m_DSCommittee) {
+    LOG_GENERAL(INFO, i.second);
+  }
+
+  if (BROADCAST_GOSSIP_MODE) {
+    std::vector<Peer> peers;
+    for (const auto& i : *m_mediator.m_DSCommittee) {
+      if (i.second.m_listenPortHost != 0) {
+        peers.emplace_back(i.second);
+      }
+    }
+    P2PComm::GetInstance().InitializeRumorManager(peers);
+  }
+
+  if (m_awaitingToSubmitNetworkInfoUpdate && GUARD_MODE) {
```

Comment:
Did you run into any issue that required adding this flag `m_awaitingToSubmitNetworkInfoUpdate`?

Reply:
no. It is more like a safety feature where i want to restrict the window for the update.

Reply:
Ok